udong-backend/udong/post/views.py

```diff
@@ -18,11 +21,59 @@
 else:
     _GenereicViewSet = viewsets.GenericViewSet
 
+_MT_co = TypeVar("_MT_co", bound=Model, covariant=True)
+
 
 class PostViewSet(_GenereicViewSet):
     queryset = Post.objects.all()
     serializer_class = CommentSerializer
 
+    def get_serializer_class(self) -> type[BaseSerializer[_MT_co]]:
+        if self.action in ("list", "retrieve"):
+            return PostBoardSerializer
+        elif self.action == "comment":
+            return CommentSerializer
+        return self.serializer_class
+
+    # TODO: Need to be optimized
+    def list(self, request: Request) -> Response:
+        user_club = UserClub.objects.filter(Q(user_id=request.user.id))
```

Comment:
이거 `Q` 안 쓰고 그냥 `filter(user__id=request.user.id)`해도 되지 않나요?

Reply:
놀랍게도 type error가 나는 관계로....ㅠ

Reply:
??? 그러네 왜지

Reply:
근데 user_id가 UserClub 모델에 있는거라 그거로 해보려했는데 잘 안됨...

Reply:
```suggestion
        user_club = UserClub.objects.filter(user__id=cast(int, request.user.id))
```
일단 이렇게 casting하는 방법이 있고

User model 관련해서 세팅 다시 해야될듯  [문서](https://docs.djangoproject.com/en/4.1/topics/auth/customizing/#substituting-a-custom-user-model)

Reply:
`user__id`가 문제가 아니라 `request.user.id` type을 제대로 못잡는 것 같아

Reply:
아하 로그인 짜면서 좀 손보면 되지 않을까?

Reply:
`AUTH_USER_MODEL` 세팅하는건 로그인 짜기 전에 해야되지 않을까 싶은데

암튼 이 PR은 approve할게요. 여기서 할 일은 아닌거같아서